public/api/getPortfolioActivityView.php

```diff
@@ -8,69 +8,117 @@
 include 'db_connection.php';
 
 $jwtArray = include 'jwtArray.php';
+$userId = $jwtArray['userId'];
 
 $doenetId = mysqli_real_escape_string($conn, $_REQUEST['doenetId']);
 
-$success = true;
-$message = '';
+$response_arr;
+$contributors = [];
+try {
 
-$sql = "
-SELECT cc.label,
-cc.courseId,
-CAST(cc.jsonDefinition as CHAR) AS json,
-p.doenetId AS 'pageDoenetId'
-FROM course_content AS cc
-LEFT JOIN pages AS p
-    ON p.containingDoenetId = cc.doenetId
-WHERE cc.doenetId = '$doenetId'
-AND cc.isPublic = '1'
-";
-$result = $conn->query($sql);
-$label = '';
-$courseId = '';
-$pageDoenetId = '';
-$json = '';
+    $sql = "
+    SELECT cc.label,
+    cc.courseId,
+    cc.isBanned,
+    CAST(cc.jsonDefinition as CHAR) AS json,
+    c.portfolioCourseForUserId,
+    c.label as courseLabel,
+    c.image,
+    c.color,
+    u.firstName,
+    u.lastName,
+    u.profilePicture
+    FROM course_content AS cc
+    LEFT JOIN course AS c
+        ON c.courseId = cc.courseId
+    LEFT JOIN user As u
+        ON u.userId = c.portfolioCourseForUserId
+    WHERE cc.doenetId = '$doenetId'
+    AND cc.isPublic = '1'
+    ";
+    $result = $conn->query($sql);
 
-if ($result->num_rows > 0) {
-    $row = $result->fetch_assoc();
-    $label = $row['label'];
-    $courseId = $row['courseId'];
-    $pageDoenetId = $row['pageDoenetId'];
-    $json = json_decode($row["json"], true);
-}
+    if ($result->num_rows > 0) {
+        $row = $result->fetch_assoc();
+        $isBanned = $row['isBanned'];
+        if ($isBanned == '1'){
+            throw new Exception("Activity has been banned.");
+        }
+        $label = $row['label'];
+        $json = json_decode($row["json"], true);
```

Comment:
It doesn't look like we are doing anything with this json, we are just reading it out of the database, decoding it into this variable and then dropping it.